app/src/main/java/org/xbmc/kore/service/PauseCallService.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 Tomer Froumin. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.xbmc.kore.service;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.telephony.TelephonyManager;
+
+import org.xbmc.kore.R;
+import org.xbmc.kore.Settings;
+import org.xbmc.kore.host.HostConnectionObserver;
+import org.xbmc.kore.host.HostManager;
+import org.xbmc.kore.jsonrpc.method.Player;
+import org.xbmc.kore.jsonrpc.type.ListType;
+import org.xbmc.kore.jsonrpc.type.PlayerType;
+import org.xbmc.kore.utils.LogUtils;
+
+/**
+ * This service maintains a notification in the notification area while
+ * something is playing, and keeps running while it is playing.
+ * This service stops itself as soon as the playing stops or there's no
+ * connection. Thus, this should only be started if something is already
+ * playing, otherwise it will shutdown automatically.
+ * It doesn't try to mirror Kodi's state at all times, because that would
+ * imply running at all times which can be resource consuming.
+ *
+ * A {@link HostConnectionObserver} singleton is used to keep track of Kodi's
+ * state. This singleton should be the same as used in the app's activities
+ */
+public class PauseCallService extends BroadcastReceiver
+        implements HostConnectionObserver.PlayerEventsObserver {
```

Comment:
Huuum, not sure about this being a BroadcastReceiver. Take a look at the documentation, particularly at the Lifecycle section to see what might go wrong: http://developer.android.com/reference/android/content/BroadcastReceiver.html